File: trabalho 1/principal.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.List;

public class Principal {
    public static void main(String[] args) {
        List<Clima> lista = new ArrayList<>();
        String nomeBase = "base.csv";

        try (BufferedReader leitor = new BufferedReader(
                new FileReader(nomeBase, StandardCharsets.UTF_8))) {
            
            String linha;
            
            while ((linha = leitor.readLine()) != null) {
                String[] dadosLinha = linha.split(",");
                
                Clima objClima = new Clima(dadosLinha[0], dadosLinha[1], dadosLinha[2], dadosLinha[3]);

                if (!lista.contains(objClima)) {
                    lista.add(objClima);
                }
            }

            if (!lista.isEmpty()) {
                Clima maior = lista.get(0); 
                for (Clima item : lista) {
                    double tempAtual = Double.parseDouble(item.temperatura);
                    double tempMaior = Double.parseDouble(maior.temperatura);

                    if (tempAtual > tempMaior) {
                        maior = item;
                    }
                }

                System.out.println("\n>>> Registro de Maior Temperatura Encontrado:");
                System.out.println(maior);
            }
            // ----------------------------------------------

        } catch (Exception e) {
            System.err.println("Ocorreu algum erro... " + e.getMessage());
        }
    }
}

File: aula 5/Python/principal.py
from clima import Clima

lista = []
nome_base = "base.csv"

try:
    leitor = open(nome_base, "r", encoding="utf-8")

    for linha in leitor:
        dados_linha = linha.split(",")
        obj_clima = Clima(dados_linha[0], dados_linha[1], dados_linha[2], dados_linha[3])

        if obj_clima not in lista:
            lista.append(obj_clima)

    leitor.close()

    contagem = {}

    for item in lista:
        if item.temperatura == "Quente":
            if item.mes not in contagem:
                contagem[item.mes] = 0
            contagem[item.mes] += 1
            
    mes_mais_quente = ""
    maior_contagem = 0

    for mes, qtd in contagem.items():
        if qtd > maior_contagem:
            maior_contagem = qtd
            mes_mais_quente = mes

    print(f"Mês mais quente: {mes_mais_quente} ({maior_contagem} vezes)")

except Exception as e:
    print("Ocorreu algum erro...", e)
